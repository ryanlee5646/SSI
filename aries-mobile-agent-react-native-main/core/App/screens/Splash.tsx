import AsyncStorage from '@react-native-async-storage/async-storage'
import { useNavigation } from '@react-navigation/core'
import { StackNavigationProp } from '@react-navigation/stack'
import React, { useMemo } from 'react'
import { Image, StyleSheet } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import { LocalStorageKeys } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { AuthenticateStackParams, Screens } from '../types/navigators'
import { Onboarding as StoreOnboardingState } from '../types/state'

const onboardingComplete = (state: StoreOnboardingState): boolean => {
  return state.didCompleteTutorial && state.didAgreeToTerms && state.didCreatePIN
}

const resumeOnboardingAt = (state: StoreOnboardingState): Screens => {
  if (state.didCompleteTutorial && state.didAgreeToTerms && !state.didCreatePIN) {
    return Screens.CreatePin
  }

  if (state.didCompleteTutorial && !state.didAgreeToTerms) {
    return Screens.Terms
  }

  return Screens.Onboarding
}
/*
  To customize this splash screen set the background color of the
  iOS and Android launch screen to match the background color of
  of this view.
*/

const Splash: React.FC = () => {
  const [, dispatch] = useStore()
  const navigation = useNavigation<StackNavigationProp<AuthenticateStackParams>>()
  const { ColorPallet } = useTheme()
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
  })
  useMemo(() => {
    async function init() {
      try {
        // await AsyncStorage.removeItem(LocalStorageKeys.Onboarding)
        const data = await AsyncStorage.getItem(LocalStorageKeys.Onboarding)
        if (data) {
          const onboardingState = JSON.parse(data) as StoreOnboardingState
          dispatch({ type: DispatchAction.ONBOARDING_UPDATED, payload: [onboardingState] })
          if (onboardingComplete(onboardingState)) {
            navigation.navigate(Screens.EnterPin)
          } else {
            // If onboarding was interrupted we need to pickup from where we left off.
            const destination = resumeOnboardingAt(onboardingState)
            // @ts-ignore
            navigation.navigate({ name: destination })
          }
          return
        }
        // We have no onboarding state, starting from step zero.
        navigation.navigate(Screens.Onboarding)
      } catch (error) {
        // TODO:(am add error handling here)
      }
    }
    init()
  }, [])

  return (
    <SafeAreaView style={styles.container}>
      <Image source={require('../assets/img/logo-large.png')} />
    </SafeAreaView>
  )
}

export default Splash
