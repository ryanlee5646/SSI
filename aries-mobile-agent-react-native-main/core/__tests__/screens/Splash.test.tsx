import { render } from '@testing-library/react-native'
import React from 'react'

import Splash from '../../App/screens/Splash'

describe('Splash Screen', () => {
  it('Renders correctly', () => {
    const tree = render(<Splash />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
