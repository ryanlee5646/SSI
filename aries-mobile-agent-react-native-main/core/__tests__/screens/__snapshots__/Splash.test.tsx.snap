// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen Renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "default": "",
      }
    }
  />
</RNCSafeAreaView>
`;
